templates/alertmanager/alertmanager-statefulset.yaml

```
@@ -175,6 +175,15 @@ spec:
             - name: gossip
               containerPort: {{ .Values.config.memberlist.bind_port }}
               protocol: TCP
+            - name: grpc
+              containerPort: {{ .Values.config.server.grpc_listen_port }}
+              protocol: TCP
+            - containerPort: 9094
+              name: gossip-clu-tcp
+              protocol: TCP
+            - containerPort: 9094
+              name: gossip-clu-udp
+              protocol: UDP
```

```suggestion
            - containerPort: {{ .Values.config.server.grpc_listen_port }}
              name: gossip-clu-tcp
              protocol: TCP
            - containerPort: {{ .Values.config.server.grpc_listen_port }}
              name: gossip-clu-udp
              protocol: UDP
```

correct this section is only metadata only and won't actually expose anything. However it does need to match with whatever is set-up in the headless service. Otherwise the service can't find the port.

---

also do we need port `9095` for something?

---

I am confused with suggested change because :
- 9095 is for grpc and all headless services are pointing to their respective port
```
cortex-alertmanager-headless     ClusterIP   None             <none>        9095/TCP    1d
cortex-distributor-headless      ClusterIP   None             <none>        9095/TCP    1d
cortex-ingester-headless         ClusterIP   None             <none>        9095/TCP    1d
cortex-query-frontend-headless   ClusterIP   None             <none>        9095/TCP    1d
cortex-store-gateway-headless    ClusterIP   None             <none>        9095/TCP    1d
```
- 9094 is when alertmanager is running in cluster mode and use gossip on 9094 to communicate (TCP + UDP). I understand this is different than `{{ .Values.config.server.grpc_listen_port }}`